File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) 
    {
        unordered_set<char> mp;
        int ans=0;
        int len=0;
        int l=0;
        for(int r=0; r<s.length(); r++)
        {

            if(mp.find(s[r])==mp.end()){
                len+=1;
                mp.insert(s[r]);
                ans=max(ans,len);
            }
            else{
                while(mp.find(s[r])!=mp.end()){
                    mp.erase(s[l]);
                    len--;
                    l++;
                }
                len+=1;
                mp.insert(s[r]);
            }
            
        }
        return ans;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) 
    {
        HashSet<Character> temp = new HashSet<>();

        int len=0;
        int j=0;
        for(int i=0; i<s.length(); i++)
        {
       
            while(j<i && (temp.contains(s.charAt(i))))
            {
                temp.remove(s.charAt(j));
                j++;
            }
                
            
            temp.add(s.charAt(i));
            len=Math.max(len, i - j + 1);
            
        }

        return len;
    }
}